lodash/lodash-tests.ts

```diff
@@ -2245,6 +2245,46 @@ namespace TestUniqBy {
     }
 }
 
+// _.uniqWith
+namespace TestUniqWith {
+    type SampleObject = {a: number; b: string; c: boolean};
+
+    let array: any[];
+    let objArray: SampleObject[];
+    let letters: string;
+    let comparator: (a: any, b : any) => boolean;
+
+    {
+        let result: any[];
+        result = _.uniqWith(array, comparator);
+    }
+
+    {
+        let result: SampleObject[];
+        result = _.uniqWith(objArray, comparator);
+    }
+
+    {
+        let result: string[];
+        result = _.uniqWith('abc', comparator);
+    }
+
+    {
+        let result: _.LoDashImplicitArrayWrapper<string[]>;
+        result = _(['abc', 'bca']).uniqWith<string[]>(comparator);
+    }
+
+    {
+        let result: _.LoDashImplicitArrayWrapper<string[]>;
+        result = _('abc').uniqWith<string[]>(comparator);
+    }
+
+    {
+        let result: _.LoDashImplicitArrayWrapper<string[]>;
+        result = _('abc').chain().uniqWith<string[]>(comparator);
```

Comment:
I believe tests shouldn't enforce typings (< type > enforces typing, not specifies it, look [at this example ](http://www.typescriptlang.org/play/#src=type%20TMyType%20%3D%20%7B%20b%3A%20string%20%7D%3B%0Afunction%20test(t%3A%20TMyType)%20%7B%20%7D%3B%0Aconst%20a%20%3D%20(%3CTMyType%3E%20%7B%7D)%3B%0Aconst%20b%20%3D%20%7B%7D%0Atest(a)%3B%0Atest(b)%3B%0A))